src/ViewModels/ArcGisMapServerItemViewModel.js

```diff
@@ -66,6 +66,8 @@ ArcGisMapServerItemViewModel.prototype._enableInCesium = function() {
 
     var scene = this.application.cesium.scene;
 
+    this.legendUrl = this.url + '/legend';
```

Comment:
Here's where the `legendUrl` property is defined in the base class:
https://github.com/NICTA/nationalmap/blob/ckanNMinit/src/ViewModels/CatalogItemViewModel.js#L62

Then this code in `CatalogMemberViewModel` will set that property from the value in the init file:
https://github.com/NICTA/nationalmap/blob/ckanNMinit/src/ViewModels/CatalogMemberViewModel.js#L188

Until `_enableInCesium` or `_enableInLeaflet` is invoked, the `legendUrl` property will have whatever value was specified in the init file.  After one of those functions is invoked, it will be set to `this.url + '/legend'`.

If you just copy/paste the `legendUrl` property code from `WebMapServiceItemViewModel`, it will work exactly as it should (at least according to me).  If `legendUrl` is not specified in the init file, its value will be `this.url + '/legend'`.  If it is specified in the init file, the value will be whatever was specified in the init file.  I don't really see how any of this is a hack.